zetaclient/chains/evm/signer/gas.go

```diff
@@ -73,6 +77,17 @@ func gasFromCCTX(cctx *types.CrossChainTx, logger zerolog.Logger) (Gas, error) {
 			Uint64("cctx.initial_gas_limit", params.CallOptions.GasLimit).
 			Uint64("cctx.gas_limit", limit).
 			Msgf("Gas limit is too high; Setting to the maximum (%d)", maxGasLimit)
+	} else if limit == gasTransferGasLimit && (params.CoinType != coin.CoinType_Gas || (cctx.IsCurrentOutboundRevert() && cctx.RevertOptions.CallOnRevert)) {
```

Comment:
Just wondering who would be paying for this increased limit, as we would have only charged the user `gasTransferGasLimit` when creating the revert outbound?

Reply:
Yes, there is an underpayment until we address https://github.com/zeta-chain/node/issues/3723 but it has been the actual behavior since the beginning until we patching with the cctx data.

Right now it is a bit mitigated in the sense it strictly check for 21k, so the user will still have to pay a fee but we need to complete this before adding the refund PR